lib/src/components/arm/client.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:grpc/grpc_connection_interface.dart';
+import 'package:viam_sdk/src/gen/common/v1/common.pb.dart';
+import 'package:viam_sdk/src/gen/component/arm/v1/arm.pbgrpc.dart';
+
+import '../../utils.dart';
+import 'arm.dart';
+
+class ArmClient extends Arm {
+  ClientChannelBase _channel;
+  ArmServiceClient _client;
+
+  ArmClient(super.name, this._channel) : _client = ArmServiceClient(_channel);
+
+  @override
+  Future<Pose> getEndPosition({Map<String, dynamic>? extra}) async {
+    final response = await _client.getEndPosition(GetEndPositionRequest(name: name));
+    return response.pose;
+  }
+
+  @override
+  Future<bool> isMoving({Map<String, dynamic>? extra}) async {
+    final response = await _client.isMoving(IsMovingRequest(name: name));
+    return response.isMoving;
+  }
+
+  @override
+  Future<void> moveToJointPositions(JointPositions positions, {Map<String, dynamic>? extra}) async {
+    await _client.moveToJointPositions(MoveToJointPositionsRequest(name: name, positions: positions));
```

Comment:
just wondering, Will we want a way to communicate that the `moveToJointPositions` was successful?